### ISSUE TITLE
Component not re-rendering on knob change.

### ISSUE THREAD

**Issue by daraclare** [storybookjs/storybook#10609]
**Describe the bug**
I think this is a bug, not a feature request 🤔 

If I change a knob that affects the design of a component, the component in storybook changes, but not if I change a knob that affects how the component renders. 

For instance, I have an iframe component (see below). If the iframe component is passed a thumbnail it does not render the iframe until the thumbnail is clicked. If he iframe component is not passed a thumbnail it should render the iframe without needing to be clicked.

I cannot currently demo this behaviour using storybook knobs 😞 

**To Reproduce**
1. Build a story with the component and story below. 
2. Run storybook
3. Change the thumbnail knob to be empty
4. Component should re-render and show the iframe, rather than show neither iframe or thumbnail until clicked

**Expected behavior**
The component should render to demonstrate how the component will render with the adjusted props.

**Code snippets**
The story:
```
import { storiesOf } from '@storybook/react';
import React from 'react';
import { withKnobs, text } from '@storybook/addon-knobs';

import { Iframe } from '../Iframe';

const WidthDecorator = (storyFn: any) => (
  <div
    style={{
      maxWidth: '640px',
    }}
  >
    {storyFn()}
  </div>
);

const stories = storiesOf('Iframe', module)
  .addDecorator(withKnobs)
  .addDecorator(WidthDecorator);

stories.add('Iframe', () => (
  <Iframe
    title={text('title', 'virtual-tour')}
    src={text('src', 'some_url')}
    thumbnail={text('thumbnails', 'https://via.placeholder.com/640x360')}
    trackingAttribute="virtual-tour"
  />
));
```

The component:
```
export const Iframe: FC<IframeProps> = ({
  title,
  src,
  thumbnail,
  trackingAttribute,
}) => {
  const [showThumbnail, setShowThumbnail] = useState(thumbnail ? true : false);

  return (
    <StyledIframeWrapper
      thumbnail={thumbnail ? thumbnail : ''}
      data-tracking={trackingAttribute}
      onClick={() => setShowThumbnail(false)}
      data-testid={`${title}-thumbnail`}
    >
      {!showThumbnail && (
        <StyledIframe
          title={title}
          src={src}
          frameBorder="0"
          allowFullScreen
          data-testid={`${title}-iframe`}
        />
      )}
    </StyledIframeWrapper>
  );
};
```
**System:**
```
Environment Info:

  System:
    OS: macOS 10.15.3
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
  Binaries:
    Node: 10.15.3 - ~/.nvm/versions/node/v10.15.3/bin/node
    Yarn: 1.19.1 - /usr/local/bin/yarn
    npm: 6.14.4 - ~/Code/dsch-frontend-toolkit-daft/node_modules/.bin/npm
  Browsers:
    Chrome: 81.0.4044.129
    Firefox: 75.0
    Safari: 13.0.5
  npmPackages:
    @storybook/addon-a11y: ^5.3.18 => 5.3.18 
    @storybook/addon-actions: ^5.3.18 => 5.3.18 
    @storybook/addon-knobs: ^5.3.18 => 5.3.18 
    @storybook/addon-links: ^5.3.18 => 5.3.18 
    @storybook/addon-viewport: ^5.3.18 => 5.3.18 
    @storybook/react: ^5.3.18 => 5.3.18 
```

**Additional context**
Storybook is great, thanks 😄

**Comment by ing200086** [comment]
I believe I am having the same/similar issue. I create a repo that can be cloned to recreate:
[repository](https://github.com/ing200086/-aiki)

Knob is used on the IncrementInput component.

If I try to update the knob it doesnt recreate the component. If I hit enter when modifying the knob it reloads the entire story, but resets the knob back to default.

**Comment by shilman** [comment]
Hi gang, We’ve just released addon-controls in 6.0-beta!

Controls are portable, auto-generated knobs that are intended to replace addon-knobs long term.

- [Controls README](https://github.com/storybookjs/storybook/tree/next/addons/controls)
- [Migration instructions](https://github.com/storybookjs/storybook/blob/next/addons/controls/README.md#how-do-i-migrate-from-addon-knobs)

Please upgrade and try them out today. Thanks for your help and support getting this stable for release!

**Comment by shilman** [comment]
For anybody who is interested in Controls but don't know where to start, I've created a quick & dirty step-by-step walkthrough to go from a fresh CRA project to a working demo. Check it out:

=> [Storybook Controls w/ CRA & TypeScript](https://gist.github.com/shilman/69c1dd41a466bae137cc88bd2c6ef487)

There are also some "knobs to controls" migration docs in the Controls README:

=> [How do I migrate from addon-knobs?](https://github.com/storybookjs/storybook/blob/next/addons/controls/README.md#how-do-i-migrate-from-addon-knobs)
